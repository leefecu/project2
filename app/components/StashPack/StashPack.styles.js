import { StyleSheet } from 'react-native';

import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

export const STACH_PACK_WIDTH = Metrics.width - Metrics.doublePadding * 2;

export default StyleSheet.create({
  container: {
    alignItems: 'stretch',
    flexDirection: 'row',
    width: STACH_PACK_WIDTH,
  },
  labelColumn: {
    alignItems: 'flex-start',
    backgroundColor: Colors.white,
    borderTopLeftRadius: Metrics.inputBorderRadius,
    borderBottomLeftRadius: Metrics.inputBorderRadius,
    flex: 6,
    justifyContent: 'center',
    paddingHorizontal: Metrics.basePadding,
    paddingVertical: Metrics.basePadding,
  },
  buttonColumn: {
    alignItems: 'center',
    backgroundColor: Colors.sunriseYellow,
    borderTopRightRadius: Metrics.inputBorderRadius,
    borderBottomRightRadius: Metrics.inputBorderRadius,
    flex: 4,
    justifyContent: 'center',
    paddingVertical: Metrics.smallPadding,
  },
  packNameBlock: {
    alignItems: 'flex-end',
  },
  packName: {
    color: Colors.primaryDark,
    fontSize: Metrics.Fonts.small,
    fontFamily: 'CircularPro-Bold',
    marginRight: Metrics.smallestMargin,
  },
  unit: {
    color: Colors.primaryDark,
    fontSize: Metrics.Fonts.small,
    fontFamily: 'CircularPro-Bold',
  },
  info: {
    borderTopColor: Colors.lineColor,
    borderTopWidth: 1,
    flexDirection: 'row',
    marginTop: Metrics.basePadding,
    paddingTop: Metrics.basePadding,
  },
  typeColumn: {
    flex: 5,
    justifyContent: 'center',
    paddingRight: Metrics.smallestPadding,
  },
  type: {
    color: Colors.descriptionColor,
    fontSize: Metrics.Fonts.smallest,
    fontFamily: 'CircularPro-Book',
  },
  remainingColumn: {
    borderLeftColor: Colors.lineColor,
    borderLeftWidth: 1,
    flex: 5,
    paddingLeft: Metrics.smallerPadding,
  },
  remaining: {
    color: Colors.descriptionColor,
    fontSize: Metrics.Fonts.smallest,
    fontFamily: 'CircularPro-Bold',
  },
});
