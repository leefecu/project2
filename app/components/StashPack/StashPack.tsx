import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';

import Button from '../Button/Button';

import Styles, { STACH_PACK_WIDTH } from './StashPack.styles';
import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

interface Props extends TStash {
  onPress: (packetId: string) => void;
}

const BUTTON_WIDTH = STACH_PACK_WIDTH * 0.4 - Metrics.basePadding * 2;

const StashPack = ({
  grantedForFree,
  onPress,
  packetId,
  packName,
  packActivationsRemaining,
}: Props) => {
  const handlePress = () => {
    onPress(packetId);
  };
  return (
    <TouchableHighlight
      activeOpacity={0.5}
      onPress={handlePress}
      underlayColor={Colors.transparent}
    >
      <View style={Styles.container}>
        <View style={Styles.labelColumn}>
          <View style={Styles.packNameBlock}>
            <Text style={Styles.packName}>{packName}</Text>
          </View>
          <View style={Styles.info}>
            <View style={Styles.typeColumn}>
              <Text style={Styles.type}>{grantedForFree ? 'Free from us!' : 'Purchased'}</Text>
            </View>
            {packActivationsRemaining && packActivationsRemaining > 1 && (
              <View style={Styles.remainingColumn}>
                <Text style={Styles.remaining}>{`${packActivationsRemaining} remaining`}</Text>
              </View>
            )}
          </View>
        </View>
        <View style={[Styles.buttonColumn]}>
          <Button
            onPress={handlePress}
            id={`Start Stash ${packName}`}
            label="Start"
            smallButton
            width={BUTTON_WIDTH}
          />
        </View>
      </View>
    </TouchableHighlight>
  );
};

StashPack.defaultProps = {
  grantedForFree: false,
};

export default StashPack;
