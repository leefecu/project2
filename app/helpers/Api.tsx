import get from 'lodash/get';
import reject from 'lodash/reject';
import Config from 'react-native-config';
import compareVersions from 'compare-versions';
import { Platform } from 'react-native';

import * as Storage from './Storage';
import { createExpiryNotification } from './Notification';
import { convertStashToActivePAck } from './Transformation';
import API from '../services/Api';

import { SYSTEM_RESPONSE } from '../constants/Shared';

const api = API.create();

export const purchasePack = async (packetId: string, promoId?: string) => {
  const response = await api.purchasePack(packetId, promoId);
  const currentActivePack = await Storage.getActivePack();
  if (response.ok) {
    const balance = get(response, 'data.balance');
    const purchasedPack = get(response, 'data.purchasedPack');
    const activePack = currentActivePack
      ? {
          ...currentActivePack,
          endTime: purchasedPack.endTime,
        }
      : purchasedPack;

    createExpiryNotification(purchasedPack.endTime);
    await Storage.setActivePackAndBalance(activePack, balance);
    return response;
  }
  return response;
};

export const activateStash = async (stash: TStash) => {
  const response = await api.activateStash(stash.packetInstanceId);
  if (response.ok) {
    const currentActivePack = await Storage.getActivePack();
    const currentStash = await Storage.getStash();
    const newActivePack = convertStashToActivePAck(stash, currentActivePack);
    const newStash = reject(
      currentStash,
      (target: TStash) => target.packetInstanceId === stash.packetInstanceId,
    );

    createExpiryNotification(newActivePack.endTime);
    await Storage.saveActivePackAndStash(newActivePack, newStash);
    return response;
  }
  return response;
};

export const checkSystemInfo = async (): Promise<SYSTEM_RESPONSE> => {
  const response = await api.getSystemInfo();
  if (response.ok) {
    const maintenance: boolean = get(response, 'data.maintenance', false);
    if (maintenance) {
      return SYSTEM_RESPONSE.MAINTENANCE;
    }
    const minVersion = get(response, `data.${Platform.OS}`);
    const appVersion = Config.APP_VERSION;

    return compareVersions.compare(appVersion, minVersion, '>=')
      ? SYSTEM_RESPONSE.CONTINUE
      : SYSTEM_RESPONSE.UPGRADE;
  }
  return SYSTEM_RESPONSE.CONTINUE;
};
