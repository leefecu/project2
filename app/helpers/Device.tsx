import { NativeModules, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import CarrierInfo from 'react-native-carrier-info';
import RNFS from 'react-native-fs';

const NZ_MCC = '530';
const NZ_COUNTRY_CODE = 'nz';
const IS_IOS = Platform.OS === 'ios';

const NetworkInfo = NativeModules.RNNetworkInfo;

export const isSimulator = async (): Promise<boolean> => {
  const isEmulator = await DeviceInfo.isEmulator();
  return isEmulator;
};

export const isSimIn = async () => {
  // react-native-carrier-info does not work in simulator
  if (await isSimulator()) {
    return true;
  }

  try {
    // mobileCountryCode method fails when no SIM card in the device
    await CarrierInfo.mobileCountryCode();
    return true;
  } catch (e) {
    return false;
  }
};

export const checkForRoaming = async (debug: boolean = false): Promise<boolean> => {
  // react-native-carrier-info does not work in simulator
  if (await isSimulator()) {
    return debug;
  }

  try {
    if (Platform.OS === 'ios') {
      const operatorPlist = await NativeModules.NetworkInfo.getOperatorPlist();
      const mcc = operatorPlist.match(/\d+\/carrier.plist/g);
      if (mcc && mcc.length > 0) {
        return mcc[0].substr(0, 3) !== NZ_MCC;
      }
      return false;
    }
    const countryCode = await NetworkInfo.isoCountryCode();
    return countryCode !== NZ_COUNTRY_CODE;
  } catch (e) {
    return false;
  }
};

const majorVersionIOS: number = parseInt(`${Platform.Version}`, 10);

export const isAutoFillSupported: boolean = IS_IOS && majorVersionIOS >= 12;

// Figured out how current(old) DataClock app stores msisdn
// it's stored in xml file
// shared_prefs/nz.co.twodegreesmobile.comptelfwd.smpl.Account.xml
const msisdnFileName = 'nz.co.twodegreesmobile.comptelfwd.smpl.Account.xml';
const msisdnFolderName = 'shared_prefs';
const msisdnRegex = /com.comptel.smpl.interaction.Account.PHONE_NUMBER_KEY">\d+<\/string/g;

export const getExistingMsisdn = async (): Promise<TMigrationData> => {
  if (IS_IOS) {
    try {
      const msisdn = await NativeModules.GetNumber.getMsisdn();
      return { number: msisdn, upgrade: !!msisdn };
    } catch (e) {
      return { number: '', upgrade: false };
    }
  }
  // This method only works with Android
  try {
    // e.g. RNFS.DocumentDirectoryPath = '/data/user/0/nz.co.twodegreesmobile.comptelfwd.smpl/files',
    const paths = RNFS.DocumentDirectoryPath.split('/');
    // need to remove "/files" and replace with "shared_prefs"
    paths[paths.length - 1] = msisdnFolderName;
    const path = paths.join('/');
    const xmlPath = `${path}/${msisdnFileName}`;
    const existingAccount = await RNFS.readFile(xmlPath);
    const match = msisdnRegex.exec(existingAccount);
    if (match && match.length > 0) {
      const msisdn = match[0]
        .replace('com.comptel.smpl.interaction.Account.PHONE_NUMBER_KEY">', '')
        .replace('</string', '');

      return { number: msisdn, upgrade: true };
    }
    return { number: '', upgrade: false };
  } catch (e) {
    return { number: '', upgrade: false };
  }
};
