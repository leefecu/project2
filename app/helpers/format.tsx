export const getFormattedMobileNumber = (number: string): string => {
  const prefix = '+64';
  const numberParts = [];
  if (number.startsWith('+64')) {
    numberParts.push(number.substring(3, 5));
    numberParts.push(number.substring(5, 8));
    numberParts.push(number.substring(8, 13));
  } else if (number.startsWith('64')) {
    numberParts[0] = number.substring(2, 4);
    numberParts.push(number.substring(4, 7));
    numberParts.push(number.substring(7, 12));
  } else if (number.startsWith('0064')) {
    numberParts[0] = number.substring(4, 6);
    numberParts.push(number.substring(6, 9));
    numberParts.push(number.substring(9, 14));
  } else if (number.startsWith('02')) {
    numberParts[0] = number.substring(1, 3);
    numberParts.push(number.substring(3, 6));
    numberParts.push(number.substring(6, 12));
  } else if (number.startsWith('2')) {
    numberParts[0] = number.substring(0, 2);
    numberParts.push(number.substring(2, 5));
    numberParts.push(number.substring(5, 12));
  }

  return `${prefix} ${numberParts.join(' ')}`;
};

// number is always in the format of 22xxxyyyy
export const getLocalFormattedMobileNumber = (number: string): string => {
  return `0${number.substring(0, 2)} ${number.substring(2, 13)}`;
};

export const parseDuration = (duration: number): number =>
  duration % 60 === 0 ? duration / 60 : duration;

export const getUnit = (duration: number, short: boolean = false): string => {
  if (duration % 60 === 0) {
    const hour = duration / 60;
    return `${short ? 'hr' : 'hour'}${hour > 1 ? 's' : ''}`;
  }
  return `${short ? 'min' : 'minute'}${duration > 1 ? 's' : ''}`;
};

const DAY = 86400;
const HOUR = 3600;
const MIN = 60;

export const gerFormatedRemainingTime = (time: number): string => {
  let remainingTime = Math.ceil(Number(time));
  let displayFormatFactor = 0;
  const timeTokens = [];
  if (remainingTime >= DAY) {
    timeTokens.push(`${Math.floor(remainingTime / DAY)}d`);
    remainingTime %= DAY;
    displayFormatFactor = DAY;
  }
  if (remainingTime >= HOUR) {
    timeTokens.push(`${Math.floor(remainingTime / HOUR)}h`);
    remainingTime %= HOUR;
    if (!displayFormatFactor) {
      displayFormatFactor = HOUR;
    }
  }
  if (displayFormatFactor <= HOUR && timeTokens.length < 2 && remainingTime >= MIN) {
    timeTokens.push(`${Math.floor(remainingTime / MIN)}m`);
    remainingTime %= MIN;
    if (!displayFormatFactor) {
      displayFormatFactor = MIN;
    }
  }
  if (displayFormatFactor <= MIN && timeTokens.length < 2 && remainingTime > 0) {
    timeTokens.push(`${remainingTime}s`);
  }
  return timeTokens.join(' ');
};
