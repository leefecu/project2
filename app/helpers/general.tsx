import { NavigationParams, NavigationScreenProp, NavigationState } from 'react-navigation';

import { ERROR_CODE } from '../constants/Messages';
import { SYSTEM_RESPONSE } from '../constants/Shared';

import * as API from '../helpers/Api';

export const noop = () => {};

interface TTestProp {
  testID: string;
  accessibilityLabel: string;
}
export const testProps = (id: string): TTestProp => ({
  testID: id,
  accessibilityLabel: id,
});

export const shouldLogout = (status: number | undefined, errorCode: ERROR_CODE): boolean =>
  status === 401 ||
  status === 403 ||
  errorCode === ERROR_CODE.NOT_REGISTERED ||
  errorCode === ERROR_CODE.INVALID_PLAN ||
  errorCode === ERROR_CODE.QD_ISSUE;

export const systemCheck = async (
  navigation: NavigationScreenProp<NavigationState, NavigationParams>,
) => {
  const systemInfo = await API.checkSystemInfo();
  if (systemInfo === SYSTEM_RESPONSE.CONTINUE) {
    return;
  }
  navigation.navigate(systemInfo === SYSTEM_RESPONSE.MAINTENANCE ? 'Maintenance' : 'Upgrade');
};
