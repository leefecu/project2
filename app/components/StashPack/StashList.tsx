import React, { PureComponent } from 'react';
import { FlatList, View } from 'react-native';
import throttle from 'lodash/throttle';

import StashPack from './StashPack';

import { THROTTLING } from '../../constants/Shared';

import { noop } from '../../helpers/general';

import Styles from './StashList.styles';

interface Props {
  stashes: Array<TStash>;
  onSelectStash: (stash: TStash) => void;
}

class StashList extends PureComponent<Props> {
  static defaultProps: Props = {
    stashes: [],
    onSelectStash: noop,
  };

  handleSelectStashThrottled: Function;

  constructor(props: Props) {
    super(props);

    this.handleSelectStashThrottled = throttle(this.handleSelectStash, THROTTLING, {
      trailing: false,
    });
  }

  keyExtractor = (item: TStash) => item.packetId;

  handleSelectStash = (item: TStash) => {
    const { onSelectStash } = this.props;
    onSelectStash(item);
  };

  renderStash = ({ item }: { item: TStash }) => {
    return (
      <View style={Styles.packRow}>
        <StashPack onPress={() => this.handleSelectStashThrottled(item)} {...item} />
      </View>
    );
  };

  render() {
    const { stashes } = this.props;
    return (
      <FlatList data={stashes} keyExtractor={this.keyExtractor} renderItem={this.renderStash} />
    );
  }
}

export default StashList;
