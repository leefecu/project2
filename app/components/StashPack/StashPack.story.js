import { Text, View } from 'react-native';
import React from 'react';

import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import StashPack from './StashPack';
import Styles from '../../Stories.styles';

storiesOf('Component | StashPack', module).add('Default', () => {
  return (
    <View style={Styles.defaultBGContainer}>
      <View style={Styles.row}>
        <Text style={Styles.title}>With remaining count</Text>
        <StashPack
          onPress={action('onPress')}
          packetId="id1"
          packetInstanceId="id1"
          packName="1 hour"
          packActivationsRemaining={3}
        />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>Free</Text>
        <StashPack
          onPress={action('onPress')}
          packetId="id2"
          packetInstanceId="id2"
          packName="2 hours"
          grantedForFree
          packActivationsRemaining={1}
        />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>Free with remaining count</Text>
        <StashPack
          onPress={action('onPress')}
          packetId="id2"
          packetInstanceId="id2"
          packName="2 hours"
          grantedForFree
          packActivationsRemaining={2}
        />
      </View>
    </View>
  );
});
