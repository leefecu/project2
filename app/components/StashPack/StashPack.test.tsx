/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import StashPack from './StashPack';

import { noop } from '../../helpers/general';

describe('StashPack component', () => {
  test('render purchased with remaining', () => {
    const button = create(
      <StashPack
        onPress={noop}
        packName="1 hour"
        packetId="id1"
        packetInstanceId="id1"
        packActivationsRemaining={3}
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('render with message', () => {
    const button = create(
      <StashPack
        grantedForFree
        onPress={noop}
        packName="1 hour"
        packetId="id2"
        packetInstanceId="id2"
        packActivationsRemaining={1}
      />,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });

  test('trigger onPress function upon press', () => {
    const onPressEvent = jest.fn();
    const component = shallow(
      <StashPack
        onPress={onPressEvent}
        packName="1 hour"
        packetId="id3"
        packetInstanceId="id2"
        packActivationsRemaining={1}
      />,
    );
    component.find('Button').simulate('press');
    expect(onPressEvent).toHaveBeenCalled();
  });
});
