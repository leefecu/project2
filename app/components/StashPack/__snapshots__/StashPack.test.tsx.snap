// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StashPack component render purchased with remaining 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "row",
        "width": 702,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#ffffff",
          "borderBottomLeftRadius": 5,
          "borderTopLeftRadius": 5,
          "flex": 6,
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#004071",
              "fontFamily": "CircularPro-Bold",
              "fontSize": 17,
              "marginRight": 2,
            }
          }
        >
          1 hour
        </Text>
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#979797",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "marginTop": 16,
            "paddingTop": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 5,
              "justifyContent": "center",
              "paddingRight": 4,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#58585a",
                "fontFamily": "CircularPro-Book",
                "fontSize": 13,
              }
            }
          >
            Purchased
          </Text>
        </View>
        <View
          style={
            Object {
              "borderLeftColor": "#979797",
              "borderLeftWidth": 1,
              "flex": 5,
              "paddingLeft": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#58585a",
                "fontFamily": "CircularPro-Bold",
                "fontSize": 13,
              }
            }
          >
            3 remaining
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffce61",
            "borderBottomRightRadius": 5,
            "borderTopRightRadius": 5,
            "flex": 4,
            "justifyContent": "center",
            "paddingVertical": 12,
          },
        ]
      }
    >
      <View
        accessibilityLabel="Start Stash 1 hour button"
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "width": 248.8,
            },
            Object {
              "alignItems": "stretch",
              "backgroundColor": "transparent",
              "borderColor": "#004071",
              "borderRadius": 32,
              "borderWidth": 1,
              "paddingVertical": 12,
            },
            Object {
              "paddingVertical": 8,
            },
            false,
            false,
            false,
            false,
          ]
        }
        testID="Start Stash 1 hour button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#004071",
                "fontFamily": "CircularPro-Bold",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "fontSize": 15,
              },
              false,
              false,
              false,
              false,
            ]
          }
        >
          Start
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`StashPack component render with message 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flexDirection": "row",
        "width": 702,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#ffffff",
          "borderBottomLeftRadius": 5,
          "borderTopLeftRadius": 5,
          "flex": 6,
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#004071",
              "fontFamily": "CircularPro-Bold",
              "fontSize": 17,
              "marginRight": 2,
            }
          }
        >
          1 hour
        </Text>
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#979797",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "marginTop": 16,
            "paddingTop": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 5,
              "justifyContent": "center",
              "paddingRight": 4,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#58585a",
                "fontFamily": "CircularPro-Book",
                "fontSize": 13,
              }
            }
          >
            Free from us!
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffce61",
            "borderBottomRightRadius": 5,
            "borderTopRightRadius": 5,
            "flex": 4,
            "justifyContent": "center",
            "paddingVertical": 12,
          },
        ]
      }
    >
      <View
        accessibilityLabel="Start Stash 1 hour button"
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "width": 248.8,
            },
            Object {
              "alignItems": "stretch",
              "backgroundColor": "transparent",
              "borderColor": "#004071",
              "borderRadius": 32,
              "borderWidth": 1,
              "paddingVertical": 12,
            },
            Object {
              "paddingVertical": 8,
            },
            false,
            false,
            false,
            false,
          ]
        }
        testID="Start Stash 1 hour button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#004071",
                "fontFamily": "CircularPro-Bold",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "fontSize": 15,
              },
              false,
              false,
              false,
              false,
            ]
          }
        >
          Start
        </Text>
      </View>
    </View>
  </View>
</View>
`;
